client/my-sites/plugins/plugin-icon/plugin-icon.jsx

```diff
@@ -14,6 +14,16 @@ export default React.createClass( {
 		isPlaceholder: React.PropTypes.bool
 	},
 
+	shouldComponentUpdate: function( nextProps ) {
+		if ( this.props.isPlaceholder !== nextProps.isPlaceholder ) {
+			return true;
+		}
+		if ( this.props.image !== nextProps.image ) {
```

Comment:
Unless we have more props that aren't documented we can probably use the pure render mixin here